Content.Server/StationEvents/Events/IonStormRule.cs

```diff
@@ -0,0 +1,234 @@
+                // hopefully work with existing glitched laws if there are multiple ion storms
+                FixedPoint2 baseOrder = FixedPoint2.New(1);
+                foreach (var law in laws.Laws)
+                {
+                    if (law.Order < baseOrder)
+                        baseOrder = law.Order;
+                }
+
+                RobustRandom.Shuffle(laws.Laws);
+
+                // change order based on shuffled position
+                for (int i = 0; i < laws.Laws.Count; i++)
+                {
+                    laws.Laws[i].Order = baseOrder + i;
+                }
+            }
+
+            // see if we can remove a random law
+            if (laws.Laws.Count > 0 && RobustRandom.Prob(target.RemoveChance))
+            {
+                var i = RobustRandom.Next(laws.Laws.Count);
+                laws.Laws.RemoveAt(i);
+            }
+
+            // generate a new law...
+            var newLaw = GenerateLaw();
+
+            // see if the law we add will replace a random existing law or be a new glitched order one
+            if (laws.Laws.Count > 0 && RobustRandom.Prob(target.ReplaceChance))
+            {
+                var i = RobustRandom.Next(laws.Laws.Count);
+                laws.Laws[i] = new SiliconLaw()
+                {
+                    LawString = newLaw,
+                    Order = laws.Laws[i].Order
+                };
+            }
+            else
+            {
+                laws.Laws.Insert(0, new SiliconLaw()
+                {
+                    LawString = newLaw,
+                    Order = -1,
+                    LawIdentifierOverride = "#"
+                });
+            }
+
+            _adminLogger.Add(LogType.Mind, LogImpact.High, $"{ToPrettyString(ent):silicon} had its laws changed by an ion storm to {laws.LoggingString()}");
+
+            // laws unique to this silicon, dont use station laws anymore
+            EnsureComp<SiliconLawProviderComponent>(ent);
+            var ev = new IonStormLawsEvent(laws);
+            RaiseLocalEvent(ent, ref ev);
+        }
+    }
+
+    // for your own sake direct your eyes elsewhere
+    private string GenerateLaw()
+    {
+        // pick all values ahead of time to make the logic cleaner
+        var threats = Pick("Threats");
+        var objects = Pick("Objects");
+        var crew1 = Pick("Crew");
+        var crew2 = Pick("Crew");
+        var adjective = Pick("Adjectives");
+        var verb = Pick("Verbs");
+        var number = Pick("NumberBase") + " " + Pick("NumberMod");
+        var area = Pick("Areas");
+        var feeling = Pick("Feelings");
+        var must = Pick("Musts");
+        var require = Pick("Requires");
+        var action = Pick("Actions");
+        var allergy = Pick("Allergies");
+        var allergySeverity = Pick("AllergySeverities");
+        var species = Pick("Species");
+        var concept = Pick("Concepts");
+        var drink = Pick("Drinks");
+        var food = Pick("Foods");
```

Comment:
put the strings on the component or at the very least validate the constants. mf dropped 15 hardcoded prototypes.